scripts/build.js

```diff
@@ -99,32 +99,48 @@ async function fetchNativeCode(url, version, path) {
 }
 
 function buildLocally() {
+    const platform = os.platform();
+    let arch = os.arch();
+
+    // Allow cross-compile (so OSX can do arm64 or x64 builds) via:
+    // --target-arch ARCH
+    if (process.argv.includes('--target-arch')) {
+        arch = process.argv[process.argv.indexOf('--target-arch') + 1];
+    }
+
+    // options for cmake.BuildSystem
     let options = {
-        CMAKE_EXPORT_COMPILE_COMMANDS: true,
-        CMAKE_JS_PLATFORM: os.platform,
-        CMAKE_JS_ARCH: os.arch,
-        BUILD_TESTING: 'OFF',
-        CMAKE_INSTALL_PREFIX: 'crt/install',
-        CMAKE_PREFIX_PATH: 'crt/install',
+        target: "install",
+        debug: process.argv.includes('--debug'),
+        arch: arch,
+        out: path.join('build', `${platform}-${arch}`),
+        cMakeOptions: {
+            CMAKE_EXPORT_COMPILE_COMMANDS: true,
+            CMAKE_JS_PLATFORM: platform,
+            BUILD_TESTING: 'OFF',
+            CMAKE_INSTALL_PREFIX: 'crt/install',
+            CMAKE_PREFIX_PATH: 'crt/install',
+        }
+    }
+
+    if (platform === 'darwin') {
+        // Node calls it "x64" but Apple calls it "x86_64", they both agree on "arm64" though
+        options.cMakeOptions.CMAKE_OSX_ARCHITECTURES = (arch === 'x64') ? 'x86_64' : arch;
```

Comment:
Just curious, this add a CMAKE_OSX_ARCHITECTURES cmake option, but the `options.arch` is still the old arch. Is that what we want?

Reply:
Is that because from the [doc](https://github.com/cmake-js/cmake-js#runtimes) from cmakejs? (I am not sure I am looking at the right place though) 
> Notice: on non-Windows systems the C++ toolset's architecture's gonna be used despite this setting. 

If so, can we add a comment about it?

Reply:
Ha, I didn't even notice that comment. I just found out cmake-js wasn't passing the right flags and knew how to fix it.
I added a comment, along with that citation